### DISCUSSION TITLE
RISC-V and 4TOPs Hardware

### DISCUSSION THREAD

**Discussion by dougransom** [blakeblackshear/frigate#8410] [Ask A Question]
Is Frigate likely to run on Risk-V boards such as this [Beagle Board](https://www.tomshardware.com/news/beagleboard-riscv-beagle-v-ahead) and leverage the 4TOPS NPU?

**Comment by fede2cr** [comment]
Currently there is no docker container that will run on riscv64 so this would be the first step.

Once that is ready, the each of the NPUs used in riscv hardware has to be added. For the beagleV ahead it would be the "HBB" collection which is T-head's fork of Apache's TVM but is hasn't got any updates in GitHub for ~2 years now.

I don't have a BeagleV but I do have quite a collection of riscv64 hardware. I plan to start with the docker image on my milkv pioneer and once that is working, we can start testing on smaller boards.

If you can help, try following the HBB tutorial from t-head, install all of the components and try running manually the models that frigate uses, and make a tutorial specific for the BeagleV with instructions on how to install it, how to run the models, and bit of benchmark would be very helpful.
